tests/unit/test_gitlab.py

```diff
@@ -353,3 +353,60 @@ def test_gitlab_plain_const_does_not_warn(recwarn):
 
     assert not recwarn
     assert no_access == 0
+
+
+@responses.activate
+@pytest.mark.parametrize(
+    "kwargs,link_header,expected_next_url,show_warning",
+    [
+        # normal execution
+        (
+            {},
+            "<http://localhost/api/v4/tests?per_page=1&page=2>;" ' rel="next"',
+            "http://localhost/api/v4/tests?per_page=1&page=2",
+            False,
+        ),
+        # got different url and will show the warning
+        (
+            {},
+            "<http://orig_host/api/v4/tests?per_page=1&page=2>;" ' rel="next"',
+            "http://orig_host/api/v4/tests?per_page=1&page=2",
+            True,
+        ),
+        # persist the base url
+        (
+            {"persist_base_url": True},
+            "<http://orig_host/api/v4/tests?per_page=1&page=2>;" ' rel="next"',
+            "http://localhost/api/v4/tests?per_page=1&page=2",
+            False,
+        ),
+    ],
+)
```

Comment:
Thanks for adding the tests @iomarmochtar! I probably would have added a separate explicit test for warnings but this works too. Instead of adding comments, we can codify each parameter into `ids` and they will also get more human-readable test names displayed at runtime during test execution/failure:

```suggestion
@pytest.mark.parametrize(
    "kwargs,link_header,expected_next_url,show_warning",
    [
        (
            {},
            "<http://localhost/api/v4/tests?per_page=1&page=2>;" ' rel="next"',
            "http://localhost/api/v4/tests?per_page=1&page=2",
            False,
        ),
        (
            {},
            "<http://orig_host/api/v4/tests?per_page=1&page=2>;" ' rel="next"',
            "http://orig_host/api/v4/tests?per_page=1&page=2",
            True,
        ),
        (
            {"persist_base_url": True},
            "<http://orig_host/api/v4/tests?per_page=1&page=2>;" ' rel="next"',
            "http://localhost/api/v4/tests?per_page=1&page=2",
            False,
        ),
    ],
    ids=["url-match-does-not-warn", "url-mismatch-warns", "url-mismatch-keeps-url"]
)
```

Reply:
done, set as suggested